app/services/send_rejection_email_to_candidate.rb

```diff
@@ -0,0 +1,45 @@
+class SendRejectionEmailToCandidate
+  attr_reader :application_choice
+
+  def initialize(application_choice:)
+    @application_choice = application_choice
+  end
+
+  def call
+    if candidate_applications.all?(&:rejected?)
+      CandidateMailer.all_applications_rejected(application_choice).deliver_later
```

Comment:
Is this deliberately different from the `application_rejected_all_rejected` email? Seems like the conditions are very similar for sending the mail but this method can handle structured reasons. I wonder if we need to consolidate the 3 possible formats somehow:

- structured reasons
- single reason (from API)
- rejected by default

This might just be me not understanding correctly, happy to chat about it.